pkg/set_test.go

```diff
@@ -0,0 +1,164 @@
+package adt
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+)
+
+func ExampleNewSet() {
+	s1 := NewSet(1, 2, 3)
+	fmt.Println(reflect.TypeOf(s1))
+
+	s2 := NewSet("a", "b", "c")
+	fmt.Println(reflect.TypeOf(s2))
+
+	s3 := NewSet[rune]() // example of initialization with 0 elements
+	fmt.Println(reflect.TypeOf(s3))
+
+	// Output:
```

Comment:
Why no asserts? Avoiding dependencies? Even if so, you can write semi-decent tests with just testing.T

Reply:
what would you assert (additionally)? i thought i'd go with testable examples which can be run as unit tests but also add to the documentation. and yes, i did not want to require anything else...

Reply:
Expected output. The problem with just printing is that you need to remember to run things and then manually inspect the output.

Reply:
This is a Go Testable Example (https://go.dev/blog/examples).

It will run this code on `go test` and assert that the printed statements match what is in the `// Ouput:` comment. So it is asserting. In addition these examples also show up in godocs.

Reply:
TIL

Reply:
i think we should get together every week and read (out loud) golang blog posts 😂